src/twister2/fixtures/dut.py

```diff
@@ -49,11 +50,21 @@ def dut(request: pytest.FixtureRequest, builder: BuilderAbstract) -> Generator[D
         hardware_map=hardware_map
     )
 
-    if not twister_config.build_only:
+    should_run = twister_config.build_only is False and spec.runnable
+    if not isinstance(request.function, YamlFunction):
+        # skip regular tests
+        if twister_config.build_only:
```

Comment:
I wondering if this skipping test execution due to `build-only` shouldn't be done in `builder` fixture. I think about this in case, when potential users wouldn't use `dut` fixture in their test and use only `builder` fixture.

Reply:
It is in `dut` fixture because of yaml tests which are executed but without flashing/running code on device. But it can be implemented in `builder` fixture, but with some small duplicated code in `dut` fixture.